apps/nycbus/nyc_bus.star

```diff
@@ -0,0 +1,178 @@
+"""
+Applet: NYC Bus
+Summary: NYC Bus departures
+Description: Real time bus departures for your preferred stop.
+Author: samandmoore
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+EXAMPLE_STOP_NAME = "21 ST/BROADWAY - SW"
+EXAMPLE_STOP_CODE = "550685"
+EXAMPLE_LOCATION = """
+{
+    "lat": "40.765302",
+    "lng": "-73.931708"
+}
+"""
+ENCRYPTED_API_KEY = "AV6+xWcEO4D+Eaj8daXafFDw3wdd6mXCdoXM4e31dPGK/CBog+fYSe07ORUjm4TdVGe4fWv5mPIbnY3dNYZNm93URlEL8KsmGsRGPTuNIP4uVnn/mvekUGBhzngbVldPA5cGY3zXt/Rbv325gs2DCbRvQUCb6qgovZMCp4WLz+/1GHn1CYmAT3zb"
+BUSTIME_STOP_TIMES_URL = "http://bustime.mta.info/api/siri/stop-monitoring.json"
+BUSTIME_STOP_INFO_URL = "http://bustime.mta.info/api/where/stop/%s.json"
+BUSTIME_STOPS_FOR_LOCATION_URL = "http://bustime.mta.info/api/where/stops-for-location.json"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.LocationBased(
+                id = "stop_code",
+                name = "Bus Stop",
+                desc = "A list of bus stops based on a location.",
+                icon = "bus",
+                handler = get_stops,
+            ),
+        ],
+    )
+
+def get_stops(location):
+    api_key = secret.decrypt(ENCRYPTED_API_KEY)
+    loc = json.decode(location)
+
+    res = http.get(
+        BUSTIME_STOPS_FOR_LOCATION_URL,
+        params = {
+            "key": api_key,
+            "lat": loc["lat"],
+            "latSpan": "0.001",
+            "lon": loc["lng"],
+            "longSpan": "0.001",
+        },
+    )
+    if res.status_code != 200:
+        fail("MTA BusTime request failed with status %d", res.status_code)
+
+    data = res.json()["data"]["stops"]
+    stops = [
+        schema.Option(display = "%s - %s" % (stop["name"], stop["direction"]), value = stop["code"])
+        for stop in data
+    ]
+
+    return stops
+
+def main(config):
+    api_key = secret.decrypt(ENCRYPTED_API_KEY) or config.get("api_key")
```

Comment:
okay, so now the tests are failing here. i think that's because the decryption doesn't work in CI?